lib/EarlyFunctionAnalysis/CFGAnalyzer.cpp

```diff
@@ -1001,8 +1006,41 @@ void CallSummarizer::handleCall(MetaAddress CallerBlock,
 
   Builder.CreateCall(PreCallHook, Args);
 
+  const auto SymbolName = extractFromConstantStringPtr(SymbolNamePointer);
+  // CalleeSummary from LatticeElement
+  const auto &[CalleeSummary, IsTail] = Oracle.getCallSite(Caller,
+                                                           CallerBlock,
+                                                           Callee,
+                                                           SymbolName);
+  const auto &ABIResults = CalleeSummary->ABIResults;
+
+  for (const auto &[CSV, Values] : ABIResults.ArgumentsRegisters) {
+    if (Values == abi::RegisterState::Yes) {
+      auto Register = M->getNamedGlobal(CSV->getName());
+      auto Name = llvm::formatv("read_{0}", CSV->getName());
+      auto F = OpaqueValuesPool.get(Name.str(),
+                                    llvm::Type::getVoidTy(Context),
+                                    { Register->getValueType() },
+                                    Name);
+      auto Load = Builder.CreateLoad(Register->getValueType(), Register);
```

Comment:
@aleclearmind could you take a look on it? Now `Load` is used in some function call. Is it enough to limit optimizing out this load?

Reply:
1. You suggested in review that it will be eliminated.
2. But in the review you told me to eliminate separate pool and put everything into one pool. Or you thought only about `write_` functions?

Reply:
1. OK, but did you check it? `:)` It was an hypothesis after some issues you mentioned on mattermost, if it's not DCE'd I wouldn't bother doing any of this.
2. The idea is to have one pool per goal: one goal is to generate opaque values to store in CSVs, a distinct goal is to prevent DCE of a load.